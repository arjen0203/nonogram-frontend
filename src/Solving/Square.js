import React from "react";
import SquareStyle from './SquareStyle.module.css';

class Square extends React.Component {
    render() {
        var display;
        if (this.props.value === 1){
            // display = "⯀";
            var filled = { backgroundColor: '#35393C'};
        } else if (this.props.value === 2){
            display = "✖";
        }
        return (
            <div
                className={SquareStyle.square}
                style={filled}
                onClick={() => this.props.onClick()}
            >
                {display}
            </div>
        );
    }
}

export default Square;