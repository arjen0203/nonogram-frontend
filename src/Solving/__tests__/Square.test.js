import React from "react";
import ReactDOM from 'react-dom';
import Square from "../Square";
import {render, cleanup} from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect"

import renderer from "react-test-renderer";

afterEach(cleanup);

it('Renders without crashing', function () {
	const div = document.createElement('div');
	ReactDOM.render(<Square></Square>, div);
});

it('Render correctly empty', function () {
	const val = 0;

	const comp = render(<Square value={val}></Square>).getByTestId("square");
	expect(comp).toBeEmpty();
	expect(comp).toHaveClass("square");
});

it('Render correctly filled in', function () {
	const val = 1;

	const comp = render(<Square value={val}></Square>).getByTestId("square");
	expect(comp).toBeEmpty();
	expect(comp).toHaveClass("square-filled");
});

it('Render correctly marked', function () {
	const val = 2;

	const comp = render(<Square value={val}></Square>).getByTestId("square");
	expect(comp).toHaveTextContent("âœ–");
	expect(comp).toHaveClass("square");
});

it('matches snapshot', function () {
	const tree = renderer.create(<Square></Square>).toJSON();
	expect(tree).toMatchSnapshot();
});