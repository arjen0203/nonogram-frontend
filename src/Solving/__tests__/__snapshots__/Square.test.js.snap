// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="square"
  data-testid="square"
  onContextMenu={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
/>
`;
